python/imzml_reader.py

```diff
@@ -139,20 +153,28 @@ def write_zarr(self, path, dtype, compressor=None, chunks=None):
         )
         # write array with metadata
         z_arr[:, :, :] = arr
-        z_arr.attrs["x_extent"] = [float(extent.x_min), float(extent.x_max)]
-        z_arr.attrs["y_extent"] = [float(extent.y_min), float(extent.y_max)]
-        z_arr.attrs["scaling_factor"] = self.ims_px_in_micro
+        self.domain = [int(np.min(z_arr)), int(np.max(z_arr))]
+        self.translate = [
+            int(extent.y_min * self.ims_px_in_micro),
+            int(extent.x_min * self.ims_px_in_micro),
+        ]
+        z_arr.attrs["domain"] = self.domain
+        z_arr.attrs["transform"] = {
+            "translate": self.translate,
+            "scale": self.ims_px_in_micro,
+        }
+
         z_arr.attrs["mz"] = self._format_mzs().tolist()
 
 
-def write_metadata_json(json_file):
-    cloud_target = open("cloud_target.txt").read().strip()
+def write_metadata_json(
+    json_file, zarr_store_url, zarr_path, domain, scale, translate
+):
     json_out = {
         "dimensions": ["mz", "y", "x"],
-        "zarrConfig": {
-            "store": f"https://s3.amazonaws.com/{cloud_target}/spraggins/",
-            "path": "spraggins.ims.zarr",
-        },
+        "zarrConfig": {"store": zarr_store_url, "path": zarr_path},
+        "domain": domain,
+        "transform": {"scale": scale, "translate": translate},
```

Comment:
It looks like you do a similar thing for z_arr.attrs... perhaps the transform could be passed around, and not just the components?